File: src/dotnet/AgentFactory/Models/Orchestration/DataSourceConfigurations/SalesforceDatacloudConfiguration.cs
using FoundationaLLM.AgentFactory.Core.Models.Orchestration.Metadata;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FoundationaLLM.AgentFactory.Core.Models.Orchestration.DataSourceConfigurations
{
    /// <summary>
    /// Blob storage configuration settings.
    /// </summary>
    public class SalesforceDatacloudConfiguration
    {
        /// <summary>
        /// The connection string key vault secret name that is retrieved from key vault.
        /// </summary>
        [JsonProperty("client_id")]
        public string? ClientId { get; set; }

        /// <summary>
        /// The name of the container
        /// </summary>
        [JsonProperty("client_secret")]
        public string? ClientSecret { get; set; }

        /// <summary>
        /// The list of files to get
        /// </summary>
        [JsonProperty("refresh_token")]
        public string? RefreshToken { get; set; }

        /// <summary>
        /// The list of files to get
        /// </summary>
        [JsonProperty("instance_url")]
        public string? InstanceUrl { get; set; }
    }
}

File: src/python/PythonSDK/foundationallm/langchain/data_sources/salesforce/salesforce_data_cloud_configuration.py
from typing import List, Optional, Any

from foundationallm.langchain.data_sources import DataSourceConfiguration

class SalesforceDataCloudConfiguration(DataSourceConfiguration):
    """Configuration settings for a connection to SalesForce DataCloud."""
    client_id: str
    client_secret: str
    refresh_token: str
    instance_url: str
    queries : List[Any]
    use_cache : bool
    columns_to_remove : Optional[List[str]]